packages/flutter/test/material/text_theme_test.dart

```diff
@@ -169,6 +169,98 @@ void main() {
+    expect(sizeTheme.displaySmall!.letterSpacing, baseTheme.displaySmall!.letterSpacing! * 2.0 + 5.0);
+    expect(sizeTheme.headlineLarge!.letterSpacing, baseTheme.headlineLarge!.letterSpacing! * 2.0 + 5.0);
+    expect(sizeTheme.headlineMedium!.letterSpacing, baseTheme.headlineMedium!.letterSpacing! * 2.0 + 5.0);
+    expect(sizeTheme.headlineSmall!.letterSpacing, baseTheme.headlineSmall!.letterSpacing! * 2.0 + 5.0);
+    expect(sizeTheme.titleLarge!.letterSpacing, baseTheme.titleLarge!.letterSpacing! * 2.0 + 5.0);
+    expect(sizeTheme.titleMedium!.letterSpacing, baseTheme.titleMedium!.letterSpacing! * 2.0 + 5.0);
+    expect(sizeTheme.titleSmall!.letterSpacing, baseTheme.titleSmall!.letterSpacing! * 2.0 + 5.0);
+    expect(sizeTheme.bodyLarge!.letterSpacing, baseTheme.bodyLarge!.letterSpacing! * 2.0 + 5.0);
+    expect(sizeTheme.bodyMedium!.letterSpacing, baseTheme.bodyMedium!.letterSpacing! * 2.0 + 5.0);
+    expect(sizeTheme.bodySmall!.letterSpacing, baseTheme.bodySmall!.letterSpacing! * 2.0 + 5.0);
+    expect(sizeTheme.labelLarge!.letterSpacing, baseTheme.labelLarge!.letterSpacing! * 2.0 + 5.0);
+    expect(sizeTheme.labelMedium!.letterSpacing, baseTheme.labelMedium!.letterSpacing! * 2.0 + 5.0);
+    expect(sizeTheme.labelSmall!.letterSpacing, baseTheme.labelSmall!.letterSpacing! * 2.0 + 5.0);
+  });
+
+  test('TextTheme apply wordSpacingFactor wordSpacingDelta', () {
+    final Typography typography = Typography.material2018();
+    final TextTheme baseTheme = Typography.englishLike2018.merge(typography.black);
+    final TextTheme baseThemeWithWordSpacing = baseTheme.copyWith(
+      displayLarge: baseTheme.displayLarge!.copyWith(wordSpacing: 1.0),
+      displayMedium: baseTheme.displayMedium!.copyWith(wordSpacing: 1.0),
+      displaySmall: baseTheme.displaySmall!.copyWith(wordSpacing: 1.0),
+      headlineLarge: baseTheme.headlineLarge!.copyWith(wordSpacing: 1.0),
+      headlineMedium: baseTheme.headlineMedium!.copyWith(wordSpacing: 1.0),
+      headlineSmall: baseTheme.headlineSmall!.copyWith(wordSpacing: 1.0),
+      titleLarge: baseTheme.titleLarge!.copyWith(wordSpacing: 1.0),
+      titleMedium: baseTheme.titleMedium!.copyWith(wordSpacing: 1.0),
+      titleSmall: baseTheme.titleSmall!.copyWith(wordSpacing: 1.0),
+      bodyLarge: baseTheme.bodyLarge!.copyWith(wordSpacing: 1.0),
+      bodyMedium: baseTheme.bodyMedium!.copyWith(wordSpacing: 1.0),
+      bodySmall: baseTheme.bodySmall!.copyWith(wordSpacing: 1.0),
+      labelLarge: baseTheme.labelLarge!.copyWith(wordSpacing: 1.0),
+      labelMedium: baseTheme.labelMedium!.copyWith(wordSpacing: 1.0),
+      labelSmall: baseTheme.labelSmall!.copyWith(wordSpacing: 1.0),
+    );
+    final TextTheme sizeTheme = baseThemeWithWordSpacing.apply(
+      wordSpacingFactor: 2.0,
+      wordSpacingDelta: 5.0,
+    );
+
+    expect(sizeTheme.displayLarge!.wordSpacing, baseTheme.displayLarge!.wordSpacing! * 2.0 + 5.0);
```

Comment:
It looks like running this line results in a `Null check operator used on a null value` error.

Reply:
I thought I ran this test to confirm it works. Guess not 🤷‍♀️.

Fixed Now.

Reply:
Awesome!

This PR is based on the [`dickermoshe:master`](https://github.com/dickermoshe/flutter/tree/master) branch, and the [commits tab](https://github.com/flutter/flutter/pull/158103/commits) isn't showing any updates since yesterday. Would you be able to push your changes to that branch?

Reply:
whoops, just pushed now.
It's one of those days...